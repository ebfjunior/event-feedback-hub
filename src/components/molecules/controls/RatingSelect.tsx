import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';

export type RatingSelectProps = {
  value?: number;
  onChange?: (rating: number | undefined) => void;
};

export function RatingSelect({ value, onChange }: RatingSelectProps) {
  return (
    <Select
      value={typeof value === 'number' ? String(value) : 'all'}
      onValueChange={(v) => onChange?.(v === 'all' ? undefined : Number(v))}
    >
      <SelectTrigger className="w-36">
        <SelectValue placeholder="All Ratings" />
      </SelectTrigger>
      <SelectContent>
        <SelectItem value="all">All Ratings</SelectItem>
        {[5, 4, 3, 2, 1].map((r) => (
          <SelectItem key={r} value={String(r)}>
            {r} stars
          </SelectItem>
        ))}
      </SelectContent>
    </Select>
  );
}


