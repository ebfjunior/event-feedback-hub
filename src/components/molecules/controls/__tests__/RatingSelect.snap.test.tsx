import React from 'react';
import { render } from '@testing-library/react';
import { RatingSelect } from '../RatingSelect';

vi.mock('@/components/ui/select', () => ({
	Select: (p: any) => <div data-mock="Select" {...p} />,
	SelectContent: (p: any) => <div data-mock="SelectContent" {...p} />,
	SelectItem: (p: any) => <div data-mock="SelectItem" {...p} />,
	SelectTrigger: (p: any) => <div data-mock="SelectTrigger" {...p} />,
	SelectValue: (p: any) => <div data-mock="SelectValue" {...p} />,
}));

describe('RatingSelect (snapshot)', () => {
	it('renders', () => {
		const { container } = render(<RatingSelect />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


