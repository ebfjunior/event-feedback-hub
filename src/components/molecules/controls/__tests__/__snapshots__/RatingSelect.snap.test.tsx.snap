// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RatingSelect (snapshot) > renders 1`] = `
<div
  data-mock="Select"
  value="all"
>
  <div
    class="w-36"
    data-mock="SelectTrigger"
  >
    <div
      data-mock="SelectValue"
      placeholder="All Ratings"
    />
  </div>
  <div
    data-mock="SelectContent"
  >
    <div
      data-mock="SelectItem"
      value="all"
    >
      All Ratings
    </div>
    <div
      data-mock="SelectItem"
      value="5"
    >
      5
       stars
    </div>
    <div
      data-mock="SelectItem"
      value="4"
    >
      4
       stars
    </div>
    <div
      data-mock="SelectItem"
      value="3"
    >
      3
       stars
    </div>
    <div
      data-mock="SelectItem"
      value="2"
    >
      2
       stars
    </div>
    <div
      data-mock="SelectItem"
      value="1"
    >
      1
       stars
    </div>
  </div>
</div>
`;
